crates/fluvio-hub-util/src/fvm/api/download.rs

```diff
@@ -43,6 +43,12 @@ pub trait Download {
 
 #[async_trait]
 impl Download for Artifact {
+    #[cfg(target_arch = "wasm32")]
+    async fn download(&self, target_dir: PathBuf) -> Result<PathBuf> {
+        unimplemented!("The download support is not implemented for wasm32 architecture");
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
```

Comment:
Since the entire fvm module is omitted for targeting wasm32, are these cfg lines needed?

Reply:
I dont think so, It could be left behind while testing on it against a WASM target locally. I could give a second check.